Library/Homebrew/utils/user.rb

```diff
@@ -0,0 +1,18 @@
+require "delegate"
+require "etc"
+
+require "system_command"
+
+class User < DelegateClass(String)
+  def gui?
+    out, _, status = system_command "who"
```

Comment:
This checks whether the user has a logged in GUI session. It does not check whether the current session is part of that GUI session.

For example, if you are logged in via the GUI and the Homebrew session is via SSH, this will return true, but AppleScript commands will still fail.

Have you tested any cases where this will return false? What are those cases?

Reply:
@reitermarkus let me know what you think of the above!

Reply:
You're right, it should check the current session, not any session. I'm not sure how to do that, however. Feel free to send a PR to improve this.